cmd/soroban-cli/src/commands/contract/build.rs

```diff
@@ -295,8 +299,54 @@ impl Cmd {
             return Ok(());
         }
 
-        let mut wasm_bytes = fs::read(target_file_path).map_err(Error::ReadingWasmFile)?;
+        // get existing wasm bytes
+        let wasm_bytes = fs::read(target_file_path).map_err(Error::ReadingWasmFile)?;
+        let existing_meta: Vec<ScMetaEntry> = Spec::new(&wasm_bytes).unwrap().meta;
+
+        let mut module = Module::new();
+
+        for payload in WasmParser::new(0).parse_all(&wasm_bytes) {
+            match payload.unwrap() {
+                Payload::CustomSection(section) => {
+                    if section.name() == META_CUSTOM_SECTION_NAME {
+                        let updated_meta = self.append_new_meta(&existing_meta).unwrap();
```

Comment:
This code assumes that the file contains only a single `contractmetav0` custom section. While that happens to be the case today during build, it's somewhat brittle to behave unexpected if that changes about how LLVM (what rustc uses to compile and construct the wasm file) builds custom sections in the future. If that changed, this logic would rewrite the existing meta multiple times. This logic would be more resilient I think if it ensure only a single meta section was writtne.